clients/drcachesim/tests/offline-phys_nopriv.templatex

```diff
@@ -0,0 +1,5 @@
+Unable to open pagemap for physical addresses: check privileges.
```

Comment:
Normally any command failure exits runmulti.cmake with a fatal error, but we need runmulti to succeed and not fail the test so we can look for this particular message.  As you saw, that's the new "failok" feature.  Are you suggesting here to have "failok" mean to not exit on a non-zero exit code, but to skip subsequent commands?  That seems a little strange to have it do that?

Reply:
> Are you suggesting here to have "failok" mean to not exit on a non-zero exit code, but to skip subsequent commands?

Not exit on a failure code (so that we can verify the expected output), and to skip the subsequent commands. 

Executing the subsequent commands doesn't really achieve anything if the previous one failed. Also, in some cases it may not be clear what the subsequent commands would behave like if the first one didn't work as expected (here it failed kind of gracefully with "failed to initialize analyzer" so probably okay for now). Also, the errors from the subsequent commands may make it less clear what the actual error was.

Reply:
> Executing the subsequent commands doesn't really achieve anything if the previous one failed. 

runmulti.cmake doesn't know that: it isn't hardocded for this particular sequence of commands, but instead has a parameter saying a non-zero exit code should be ignored.  That can easily mean that what matters is some side effect and not the final exit code, and that subsequent commands can and perhaps must be executed as they perform vital cleanup.  I.e., runmulti is a general script, and these particular properties of these drcachesim tests do not represent other test command sequences.

Reply:
> in some cases it may not be clear what the subsequent commands would behave like if the first one didn't work as expected

That's up to the caller of runmulti, who is passing these commands: arguably not up to runmulti itself.

I would think the cleanest way to avoid running the other commands, if that's what we want to do, would be to either A) not have any further commands for this particular test, and leave "failok" meaning just ignore all exit codes and still run all commands in runmulti; or B) change this "failok" to an explicit "short_circuit_on_failure" meaning to skip all further commands (but still remain inside runmulti to check the output: can't just return) on a failure in any one command.

Reply:
I set the _nopost variable to implement A.  This seems the most general and flexible: user doesn't set post-commands if they expect failure.